### PR TITLE
Update VCPKG to 2021.05.12.

### LINKED ISSUE #4873: Win86 docker image fails to build on Azure
See here: https://dev.azure.com/PointCloudLibrary/pcl/_build?definitionId=11
The Win64 docker builds successfully.
The problem is at this command, but there is no error message:
```
RUN cd .\vcpkg;                                                `
    .\bootstrap-vcpkg.bat;                                            `
    .\vcpkg install boost flann eigen3 qhull vtk[qt,opengl] gtest benchmark --triplet $Env:PLATFORM-windows-rel --clean-after-build;
```

### PR DESCRIPTION
**Pull request by larshg**
Hopefully it fixes #4873

### CODE REVIEW

**Comment by kunaltyagi** [comment]
CI still failing

**Review by mvieth — DISMISSED** [review]
Docker build succeeded, so :+1:

**Comment by larshg** [comment]
Well. I justed tested using master of VCPKG. The intention with x86 build was to use a fixed checkout. 

I'll open an issue on VCPKG or at least figure out which latest version is required.

**Comment by kunaltyagi** [comment]
Their master branch CI is red, so I don' think they really notice that 😆

**Comment by larshg** [comment]
Should we limit the cmake version as in last commit?
Or should we upgrade VCPKG to a recent commit that handles Cmakes new output -> boost 1.75->1.76, and some other minor libs that would be a new version as well.
To have the most spread in checks I guess we should limit CMake to 3.20.5?

File: .dev/docker/windows/Dockerfile [hunk 1]
```diff
@@ -34,8 +34,9 @@ RUN wget $Env:CHANNEL_BASE_URL/vs_buildtools.exe -OutFile 'C:\TEMP\vs_buildtools
     -Wait -PassThru;                                                `
     del c:\temp\vs_buildtools.exe;                                  
 
-RUN iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'));        `
-    choco install cmake git --installargs 'ADD_CMAKE_TO_PATH=System' -y --no-progress
+RUN iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'));   `
+    choco install cmake --version=3.20.5 --installargs 'ADD_CMAKE_TO_PATH=System' -y --no-progress; `
```
Comment (kunaltyagi) [inline .dev/docker/windows/Dockerfile:38]: Should we add a comment for choosing specific version of cmake?

**Review by kunaltyagi — DISMISSED** [review]

File: .dev/docker/windows/Dockerfile (hunk 1 above)
Comment (larshg) [reply .dev/docker/windows/Dockerfile:39]: Have done so.

File: .dev/docker/windows/Dockerfile [hunk 2]
```diff
@@ -34,8 +34,10 @@ RUN wget $Env:CHANNEL_BASE_URL/vs_buildtools.exe -OutFile 'C:\TEMP\vs_buildtools
     -Wait -PassThru;                                                `
     del c:\temp\vs_buildtools.exe;                                  
 
-RUN iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'));        `
-    choco install cmake git --installargs 'ADD_CMAKE_TO_PATH=System' -y --no-progress
+# VCPKG requires update if Cmake version is > 3.20.5 see: https://github.com/microsoft/vcpkg-tool/pull/107
```
Comment (kunaltyagi) [inline .dev/docker/windows/Dockerfile:37]: That PR is included in https://github.com/microsoft/vcpkg-tool/releases/tag/2021-07-16

Comment (larshg) [reply .dev/docker/windows/Dockerfile:37]: Yes, but the commit of https://github.com/microsoft/vcpkg that we are using, doesn't download that new VCPKG executable. We can update VCPKG, but then we'll also update some of the libraries (ie. boost and some other minor sublibaries).
So to have a larger span on tested libraries we "paused" the x86 on a specific commit.

**Review by kunaltyagi — APPROVED** [review]

**Review by mvieth — APPROVED** [review]

**Pull request merged**